### PR TITLE
Fix AppImage icons

### PR DESCRIPTION
**Pull request by leoheck**
The goal here is to fix the AppImage icons that are not complying with the standards and also they are not working since a long time ago.

The latest discussions here #14662
There is also an old discussion here #12915

### CODE REVIEW

**Comment by leoheck** [comment]
I beleive this would fix the issue. 

```python
    copied_files = {
        os.path.join("..", "icons", "cura-icon.svg"):         "/usr/share/icons/hicolor/scalable/apps/",
        os.path.join("..", "icons", "cura-icon_64x64.png"):   "/usr/share/icons/hicolor/64x64/apps/cura-icon.png",
        os.path.join("..", "icons", "cura-icon_128x128.png"): "/usr/share/icons/hicolor/128x128/apps/cura-icon.png",
        os.path.join("..", "icons", "cura-icon_256x256.png"): "/usr/share/icons/hicolor/256x256/apps/cura-icon.png",
        "cura.appdata.xml": "cura.appdata.xml",
        "AppRun": "AppRun"
    }
```

But I was not able to generate the AppImage to verify this myself. Any hints?

```bash
Cura/packaging/AppImage on  fix-appimage-icons [!] …
➜ python create_appimage.py cura 1.0.0 cura.AppImage
Copying /home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/../icons/cura-icon.svg to cura/usr/share/icons/hicolor/scalable/apps/
Copying /home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/../icons/cura-icon_64x64.png to cura/usr/share/icons/hicolor/64x64/apps/cura-icon.png
Copying /home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/../icons/cura-icon_128x128.png to cura/usr/share/icons/hicolor/128x128/apps/cura-icon.png
Copying /home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/../icons/cura-icon_256x256.png to cura/usr/share/icons/hicolor/256x256/apps/cura-icon.png
Copying /home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/cura.appdata.xml to cura/cura.appdata.xml
Copying /home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/AppRun to cura/AppRun
Changing permissions for AppRun
Traceback (most recent call last):
  File "/home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/create_appimage.py", line 84, in <module>
    build_appimage(args.dist_path, args.version, args.filename)
  File "/home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/create_appimage.py", line 17, in build_appimage
    copy_metadata_files(dist_path, version)
  File "/home/lheck/Dropbox/Documents4/Cura/packaging/AppImage/create_appimage.py", line 52, in copy_metadata_files
    os.chmod(os.path.join(dist_path, "AppRun"), stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH | stat.S_IXOTH)
FileNotFoundError: [Errno 2] No such file or directory: 'cura/AppRun'
```

**Comment by leoheck** [comment]
Alright, I believe I have it working. 

What I had to do to check this:

```bash
cd Cura/packaging/AppImage
APPIMAGETOOL_LOCATION=/home/lheck/Applications/appimagetool \
ARCH=x86_64 python create_appimage.py ./mycura 1.0.0 mycura.AppImage
```

The resulting AppImage now has the icon (the one on the right)
The one on the left is my previous workaround to fix the icon myself. 
image

The result checking the resulting AppImage  with appimagelint:
```bash
Cura/packaging/AppImage on  fix-appimage-icons [!?] …
➜ appimagelint-x86_64.AppImage mycura.AppImage 
appimagelint.cli[351828] [INFO] Checking AppImage mycura.AppImage
appimagelint.cli[351828] [INFO] Running check "GNU libc ABI check"
appimagelint.glibc_abi_check[351828] [INFO] detected required version for runtime: 2.14
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Debian oldstable (buster)
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Debian stable (bullseye)
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Debian testing (bookworm)
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Debian unstable (sid)
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Ubuntu kinetic
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Ubuntu jammy
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Ubuntu focal
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Ubuntu bionic
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Ubuntu xenial
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on Ubuntu trusty
appimagelint.glibc_abi_check[351828] [INFO] [✔] AppImage can run on CentOS 7
appimagelint.cli[351828] [INFO] Running check "GNU libstdc++ ABI check"
appimagelint.glibcxx_abi_check[351828] [INFO] detected required version for runtime: <none>
appimagelint.glibcxx_abi_check[351828] [WARNING] could not find any dependencies, skipping check
appimagelint.cli[351828] [INFO] Running check "Icons validity and location check"
appimagelint.icons_check[351828] [INFO] Extracting icon name from desktop file: /tmp/.mount_mycuraHMmZln/cura.desktop
appimagelint.icons_check[351828] [INFO] Checking resolution of icon: /tmp/.mount_mycuraHMmZln/cura-icon.png
appimagelint.icons_check[351828] [INFO] [✔] Valid icon in AppDir root
appimagelint.icons_check[351828] [INFO] Checking resolution of icon: /tmp/.mount_mycuraHMmZln/.DirIcon
appimagelint.icons_check[351828] [INFO] [✔] Valid icon file in .DirIcon
appimagelint.icons_check[351828] [INFO] Checking resolution of icon: /tmp/.mount_mycuraHMmZln/usr/share/icons/hicolor/128x128/apps/cura-icon.png
appimagelint.icons_check[351828] [INFO] Checking resolution of icon: /tmp/.mount_mycuraHMmZln/usr/share/icons/hicolor/256x256/apps/cura-icon.png
appimagelint.icons_check[351828] [INFO] Checking resolution of icon: /tmp/.mount_mycuraHMmZln/usr/share/icons/hicolor/64x64/apps/cura-icon.png
appimagelint.icons_check[351828] [INFO] Checking resolution of icon: /tmp/.mount_mycuraHMmZln/usr/share/icons/hicolor/scalable/apps/cura-icon.svg
appimagelint.icons_check[351828] [INFO] [✔] Other integration icons valid
appimagelint.cli[351828] [INFO] Running check "Desktop files existence and validity"
appimagelint.desktop_files[351828] [INFO] Checking desktop files in root directory
appimagelint.desktop_files[351828] [INFO] [✔] Exactly one desktop file in AppDir root
appimagelint.desktop_files[351828] [INFO] Checking desktop file /tmp/.mount_mycura6mgkNE/cura.desktop with desktop-file-validate
appimagelint.desktop_files[351828] [INFO] [✔] All desktop files in AppDir are valid

```

Hey @TheAssassin could you please check or at least give me your wise words if this is good enough for you?

File: packaging/AppImage/create_appimage.py [hunk 1]
```diff
@@ -30,15 +30,23 @@ def copy_metadata_files(dist_path, version):
     Copy metadata files for the metadata of the AppImage.
     """
     copied_files = {
+        os.path.join("..", "icons", "cura-icon.svg"):         "usr/share/icons/hicolor/scalable/apps/cura-icon.svg",
+        os.path.join("..", "icons", "cura-icon_64x64.png"):   "usr/share/icons/hicolor/64x64/apps/cura-icon.png",
+        os.path.join("..", "icons", "cura-icon_128x128.png"): "usr/share/icons/hicolor/128x128/apps/cura-icon.png",
+        os.path.join("..", "icons", "cura-icon_256x256.png"): "usr/share/icons/hicolor/256x256/apps/cura-icon.png",
```
Comment (jellespijker) [inline packaging/AppImage/create_appimage.py:36]:
```suggestion
{
        os.path.join("..", "icons", "cura-icon.svg"):         os.path.join("usr", "share", "icons", "hicolor", "scalable", "apps", "cura-icon.svg"),
        os.path.join("..", "icons", "cura-icon_64x64.png"):   os.path.join("usr", "share", "icons", "hicolor", "64x64", "apps", "cura-icon.png"),
        os.path.join("..", "icons", "cura-icon_128x128.png"): os.path.join("usr", "share", "icons", "hicolor", "128x128", "apps", "cura-icon.png"),
        os.path.join("..", "icons", "cura-icon_256x256.png"): os.path.join("usr", "share", "icons", "hicolor", "256x256", "apps", "cura-icon.png"),
```

It is always good practice to write path handling in a system agnostic way

File: packaging/icons/cura-icon.svg
Comment (jellespijker) [inline packaging/icons/cura-icon.svg:1]: This is a old icon, I asked our UXer @lorenzoromagnoli for a new svg

**Review by jellespijker — CHANGES_REQUESTED** [review]
Good initiative.

Although the svg file still uses the old icon, I asked our UXer for a new svg.

File: packaging/icons/cura-icon.svg
Comment (jellespijker) [reply packaging/icons/cura-icon.svg:1]: cura-icon

**Review by jellespijker — APPROVED** [review]

**Comment by jellespijker** [comment]
@leoheck thanks for your contribution, I will add the icon once merged to `main`

**Comment by leoheck** [comment]
Awesome, thank you @jellespijker

**Pull request merged**